kernel-api/src/main/scala/org/apache/toree/interpreter/Interpreter.scala

```diff
@@ -110,6 +110,13 @@ trait Interpreter {
    */
   def completion(code: String, pos: Int): (Int, List[String] )
 
+
+  /**
+    * Attempt to determine if a multiline block of code is complete
+    * @param code The code to determine for completeness
+    */
+  def isComplete(code: String): (String, String)
```

Comment:
We could just define a default concrete implementation in the trait and not need the companion class, your call.